fedimint-api/src/db/notifications.rs

```diff
@@ -0,0 +1,154 @@
+use std::{
+    collections::hash_map::DefaultHasher,
+    hash::{Hash, Hasher},
+};
+
+use bitvec::vec::BitVec;
+use tokio::sync::{futures::Notified, Notify};
+
+// arbitrary number. if you have performance problems, tweak this
+const NOTIFY_BUCKETS: usize = 32;
+
+/// Manages notifications for the database
+#[derive(Debug)]
+pub struct Notifications {
+    slots: Vec<Notify>,
+}
+
+impl Default for Notifications {
+    fn default() -> Self {
+        Self {
+            slots: (0..NOTIFY_BUCKETS).map(|_| Notify::new()).collect(),
+        }
+    }
+}
+
+impl Notifications {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Waits untils the key *might* have changed.
```

Comment:
Does the client of this API need to handle false positives?

Reply:
Yes, like await_key does.